src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/SasTokenGeneratorService.java

```diff
@@ -55,13 +56,9 @@ public String generateSasToken(String serviceName) {
     private SharedAccessBlobPolicy createSharedAccessPolicy(String serviceName) {
         TokenConfig config = getTokenConfigForService(serviceName);
 
-        Calendar cal = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
-        cal.setTime(new Date());
-        cal.add(Calendar.SECOND, config.getValidity());
-
         SharedAccessBlobPolicy policy = new SharedAccessBlobPolicy();
-        policy.setPermissions(getBlobPermissions());
-        policy.setSharedAccessExpiryTime(cal.getTime());
+        policy.setPermissions(EnumSet.of(WRITE, LIST));
+        policy.setSharedAccessExpiryTime(Date.from(now().plusSeconds(config.getValidity()).toInstant(ZoneOffset.UTC)));
```

Comment:
I think `LocalDateTime` should also have `.toInstant` method so `Date.from` becomes irrelevant here ;)

Reply:
`.toInstant` **is** called on LocalDateTime.
There's a lot of parentheses in this line...